source/full_shutdown.rst

```diff
@@ -0,0 +1,205 @@
+
+   .. code-block:: bash
+
+      ceph fs set FS_NAME max_mds 1
+      ceph mds deactivate FS_NAME:1 # rank 2 of 2
+      ceph status # wait for rank 1 to finish stopping
+      ceph fs set FS_NAME cluster_down true
+      ceph mds fail FS_NAME:0
+
+   Setting the cluster_down flag prevents standbys from taking over the failed rank.
+
+   - Set the noout, norecover, norebalance, nobackfill, nodown and pause flags.
+
+   .. code-block:: bash
+
+      ceph osd set noout
+      ceph osd set norecover
+      ceph osd set norebalance
+      ceph osd set nobackfill
+      ceph osd set nodown
+      ceph osd set pause
+
+   - Shut down the OSD nodes one by one:
+
+   .. code-block:: bash
+
+      systemctl stop ceph-osd.target
+
+   - Shut down the monitor/manager nodes one by one:
+
+   .. code-block:: bash
+
+      systemctl stop ceph.target
+
+Set Bifrost maintenance mode
+----------------------------
+
+Set maintenance mode in bifrost to prevent nodes from automatically
+powering back on
+
+.. code-block:: bash
+
+   for i in `openstack --os-cloud bifrost baremetal node list -c UUID -f value` ; \ 
+   do openstack --os-cloud bifrost baremetal node maintenance set $i ; done
+
+Shut down nodes
+---------------
+
+Shut down nodes one at a time gracefully using:
+
+.. code-block:: bash
+
+   systemctl poweroff
+
+Shut down the seed VM
+---------------------
+
+Shut down seed vm on ansible control host gracefully using:
+
+.. code-block:: bash
+   :substitutions:
+
+   ssh stack@|seed_name| sudo systemctl poweroff
+   virsh shutdown |seed_name|
+
+
+Full Power on Procedure
+-----------------------
+
+* Start ansible control host and seed vm
+* Remove nodes from maintenance mode in bifrost
+* Recover MariaDB cluster
+* Start Ceph (if applicable)
+* Check that all docker containers are running
+* Check Kibana for any messages with log level ERROR or equivalent
+
+Start Ansible Control Host
+--------------------------
+
+The Ansible control host is not enrolled in Bifrost and will have to be powered
+on manually.
+
+Start Seed VM
+-------------
+
+The seed VM (and any other service VM) should start automatically when the seed
+hypervisor is powered on. If it does not, it can be started with:
+
+.. code-block:: bash
+
+   virsh start seed-0
+
+Unset Bifrost maintenance mode
+------------------------------
+
+Unsetting maintenance mode in bifrost should automatically power on the nodes
+
+.. code-block:: bash
+
+   for i in `openstack --os-cloud bifrost baremetal node list -c UUID -f value` ; \
+   do openstack --os-cloud bifrost baremetal node maintenance unset $i ; done
+
+Recover MariaDB cluster
+-----------------------
+
+If all of the servers were shut down at the same time, it is necessary to run a
+script to recover the database once they have all started up. This can be done
+with the following command:
+
+.. code-block:: bash
+
+   kayobe# kayobe overcloud database recover
```

Comment:
Wondering if it would be cleaner to stop the containers before shutdown, to avoid them starting up in a broken state.